test/stress/generate-report/download_results.py

```
@@ -0,0 +1,132 @@
+        import certifi
+        ctx.load_verify_locations(certifi.where())
+    except ImportError:
+        pass
+    return ctx
+
+SSL_CONTEXT = make_ssl_context()
+
+def download_file(url, out_path):
+    print(f"Downloading {url} ...")
+    try:
+        with urllib.request.urlopen(url, timeout=60, context=SSL_CONTEXT) as response:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)